src/app/tosBanner/state/tos-banner.service.ts

```diff
@@ -33,4 +38,23 @@ export class TosBannerService {
       };
     });
   }
+
+  acceptTOS() {
+    this.userQuery.user$.pipe(takeUntil(this.ngUnsubscribe)).subscribe((user) => {
```

Comment:
this takeUntil also doesn't do anything. also, this will cause multiple unclosed subscription. need to get the user value directly from the store. search "getValue()" in the code for an example.